lib/collection/src/grouping/aggregator.rs

```diff
@@ -0,0 +1,380 @@
+use std::collections::{HashMap, HashSet};
+use std::hash::Hash;
+
+use segment::types::{ExtendedPointId, ScoredPoint};
+use serde_json::Value;
+use AggregatorError::*;
+
+pub(super) enum AggregatorError {
+    GroupsFull,
+    Other,
+}
+
+/// Abstraction over serde_json::Value to be used as a key in a HashMap/HashSet
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub(super) struct GroupKey(pub serde_json::Value);
+
+impl TryFrom<serde_json::Value> for GroupKey {
+    type Error = AggregatorError;
+
+    /// Only allows Strings and Numbers to be converted into GroupKey
```

Comment:
TBH I am still not entirely sure it should be restricted.
We could simply hash the whole `serde_json::Value` right?

Not forcing you to change it now but I like to see a unit test showing the current behavior, i.e ignoring it.

Reply:
Value doesn't implement Hash, so the other method would be to serialize it for the other types, then hash over the string version. 
I was thinking that it would be weird to use the other variants as group key, should we allow it then?

Reply:
Ah yes, the other reason is because Match Any (which is used for including/excluding groups in the fetching stage) only allows strings and numbers. Allowing the other variants would need to use the Match Value condition for constructing the internal queries

Reply:
Added a test case in `it_adds_points()` to validate this